integration/sql/iface-java/src/main/java/io/openlineage/sql/ExtractionError.java

```diff
@@ -0,0 +1,58 @@
+/*
+/* Copyright 2018-2022 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.sql;
+
+import java.util.Optional;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+public class ExtractionError {
+  private final long index;
+  private final String message;
+  private final String originStatement;
+
+  public ExtractionError(long index, String message, String originStatement) {
```

Comment:
Isn't it more `SqlParserError` than `ExtractionError`. 
In Java there is a convention that `Errors` extend `Error` class which is not the case here.

Reply:
Use of this facet isn't limited to SQL tbh. Maybe this could be moved to client as it's in Python.